### DISCUSSION TITLE
Fotek PID + Artisan Scope

### DISCUSSION THREAD

**Discussion by erestrepo12** [artisan-roaster-scope/artisan#594] [Q&A] [answered]
Hello everyone. during the pandemic I just jump into a personal projectbof making my own coffee roaster. I made nearly everything I hoped to, but there are 2 big things missing, because I tried everything I could, but still didn't make it. yettt, so I come her for help from all the good people on GitHub.
the missing things to solve are:
1. usb connection with artisan scope 2.4.4. using fotek pid with rs485 output (modbus ascii or binary signals) + usb/rs485 adapter and cable
2. I would like to add an analogue differential pressure gauge to monitor the airflow

the unit I made is totally made of recicled metals, used items and passion, the capacity for about 600g green beans, I just got mad doing it and installed all I could for safety matters, including electric gas valve, panic boton, gas manometer, pilot burner with ceramic infrared burner, 2 Temp PIDs, induction drum motor with control and tachometer, cyclone collector and control, cooling fan tray. I tested successfully many times yet, it roasted fine using a handheld datalogger and Artisan.
of there is someone who could help me please, I will be grateful for many coffee cups ahead

**Comment by roasterdave** [comment]
Congratulations on your roaster build.  Thanks for using only recycled materials!


1 - Artisan supports any device that implements MODBUS and provides its temperature data via a MODBUS register.  However, the MODBUS register addresses are not consistent from one device to the next and expecially from manufacturer to another.  

You need to find the user manual for your specific PID.  The serial parameters must be set the same in Artisan and on the PID.  These settings may be configurable from the front panel of the PID.  In Artisan they can be found at Config> Port> Modbus.

Next set the Slave ID (probably 1 for the first device).  Once the serial parameters and slave id are properly set you should be able to connect to your device.  However you are not ready yet to read data.  The register(s) to read temperature data from need to be configured in Artisan.

Once connected you can optionally use the "Scan" button to help locate data holding registers in the connected device.  You will still need your device's manual to identify the correct registers and to properly configure the MODBUS address information in Artisan.

Some additional information about MODBUS setups using other PIDs on the [Artisan Blog](https://artisan-roasterscope.blogspot.com/search?q=modbus).


2 - You mention an analogue differential pressure gauge.  Do you want to connect it to Artisan, in which case you will need a digital device.  Many people have had success using the Phidget 1136.  Setup information can be found beginning on page 8 in [this guide](https://bit.ly/RoasterUpgrade).  More information is available on the [Artisan Blog](https://artisan-roasterscope.blogspot.com/search?q=1136). 

The other option is to find an analogue device with a digital sender to get the data to Artisan.  You will need to research what is available and how to make the connection.

Please report how it goes.

**Reply by erestrepo12** [reply]
Hello Dave,  thanks for your support...  i am very greatful  for it.
I am still strugling with modbus setting (goal-1).  As all parts in my roaster project, all the items are used and ussually very  old versions, i asume the manual wasnever meaned for Artisan or anything close to it.  I am in the game of guessing register numbers and settings. 
the fotek MT-72 is set for BCD code, 8N1, i checked it can be readed with function 3 and modbus rtu

**Comment by roasterdave** [comment, marked as answer]
The datasheet for the [Fotek MT series controllers](https://www.fotek.com.tw/en-gb/download/54) indicates that it supports RS-485 communication but the communication protocol described is not MODBUS.  It appears to be a custom protocol.   A datasheet for the MT series from 2002 shows the same custom protocol.  Curious how you confirmed the MT-72 supports MODBUS RTU?

Without MODBUS support you need to write a script to implement this custom protocol to connect the MT-72 to Artisan.  Adding this protocol to Artisan is not on the roadmap, sorry.

If you are able to find one, the [Fotek NT series controllers](https://www.fotek.com.tw/en-gb/download/53) do show support of the MODBUS protocol over RS-485.  These controllers, or any controller supporting MODBUS should work directly with Artisan.
